### PR TITLE
profile update

### PR DESCRIPTION
**Pull request by DawoodMehmood**
# Pull Request Template

Before submitting this PR, please make sure you have completed the following checklist. This helps ensure that our codebase remains high quality and that our features work well across all supported devices and themes.

### Code Review
- [x] I have carefully reviewed my code and believe it is final from my side.
- [x] I have ensured there is no hardcoded color in my changes. Colors should be defined in theme files and reused.

### Theme Compatibility
- [ ] I have tested the new feature or updates in both light and dark themes.
- [ ] I confirm that the visual aspects of the feature are compatible and appealing in both themes.

### Device Compatibility
- [ ] I have tested my changes on mobile screen sizes.
- [ ] I have tested my changes on tablet screen sizes.
- [ ] I have tested my changes on laptop screen sizes.

### Functional Testing
- [x] I have thoroughly tested the functionality of my changes.
- [x] I am confident that my changes do not negatively impact existing features.

### Testing Documentation
Please provide a brief description of the tests you have performed to verify your changes. Include any relevant scenarios or edge cases.

- Example test 1: Tested feature X by uploading the image
- Example test 2: Tested feature X by doing these things.

Test #1:  Tested space updation by adding an admin 
Test #2:  Tested space updation by removing an admin 

# Demo Video
Please provide a demo video of your changes. This helps reviewers understand the changes and verify the functionality.

You can record the video using tools like Loom(https://www.loom.com/), Clip by ClickUp(https://clickup.com/features/clips), Sendspark(https://www.sendspark.com/), etc.

Include the link to the video below and make sure the video covers

https://www.loom.com/share/e8bb82f109524981aceb0668f3f7b472?sid=33221f16-6063-45e8-bbb0-7c86deb31784

- [ ] Explain the task/bug you are solving
- [ ] Show the changes in the UI
- [ ] Show the changes in the codebase
- [ ] Show the changes you tested to make sure it works
- [ ] Show the changes in various screen sizes
- [ ] Show the changes in both light and dark themes

### CODE REVIEW

File: shared/web-core/src/components/core/badge/UpsertKeyValueBadgeInput.tsx [hunk 1]
```diff
@@ -11,6 +11,8 @@ export interface UpsertBadgeItemsProps {
   className?: string;
   helpText?: string;
   inputPlaceholder?: string;
+  currentUsername: string;
```
Comment (RobinNagpal) [inline shared/web-core/src/components/core/badge/UpsertKeyValueBadgeInput.tsx:14]: A badge doesn't know what is a username. Pass some generic props

File: academy-ui/src/app/api/[spaceId]/actions/spaces/finish-space-setup/route.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
+import { NextRequest, NextResponse } from 'next/server';
```
Comment (RobinNagpal) [inline academy-ui/src/app/api/[spaceId]/actions/spaces/finish-space-setup/route.ts:1]: Can't we have used upsert Space? i.e. a PUT at `academy-ui/src/app/api/[spaceId]/spaces/[spaceId]/route.ts`. Do we need a special route for this one.

There is already a `academy-ui/src/app/api/spaces/update-space/route.ts` we should 
1. move it to  `academy-ui/src/app/api/[spaceId]/spaces/[spaceId]/route.ts`
2. use PUT
3. Change the existing place where `app/api/spaces/update-space` was used and use the new route instead.
4. Use that route for Finish Space setup also as we are just essentially updating the space

Comment (RobinNagpal) [reply academy-ui/src/app/api/[spaceId]/actions/spaces/finish-space-setup/route.ts:1]: As discussed
1. This code can go to `academy-ui/src/app/api/[spaceId]/spaces/route.ts` as PUT, becauase its an update
2. The code that we have at `cademy-ui/src/app/api/[spaceId]/spaces/[spaceId]/route.ts` can go to path `academy-ui/src/app/api/[spaceId]/actions/spaces/update-space-and-integration/route.ts`

File: academy-ui/src/app/api/[spaceId]/actions/spaces/route.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    },
+    where: {
+      id: spaceInput.id,
+    },
+  });
+  revalidateTag(SpaceTags.GET_SPACE.toString());
+  return NextResponse.json(updatedSpace as Space, { status: 200 });
+}
+
+export const PUT = withErrorHandlingV1<Space>(putHandler);
```
Comment (RobinNagpal) [inline academy-ui/src/app/api/[spaceId]/actions/spaces/route.ts:68]: This doesn't need to be actions it can be in `academy-ui/src/app/api/[spaceId]/spaces/route.ts`
